src/index.c

```diff
@@ -681,6 +681,15 @@ static void II_Rewind(void *ctx) {
   }
 }
 
+typedef int (*CompareFunc)(const void *a, const void *b);
+static int cmpIter(IndexIterator **it1, IndexIterator **it2) {
+  if (!*it1 && !*it2) return 0;
```

Comment:
a little assert for `it1 && it2`?

Reply:
These pointers are in the array passed to `qsort`. We might want to check that array earlier but here it cannot be NULL.

Reply:
Are we completely sure the array cannot contain null elements? Dereferencing pointers without an assert causes me an itch.

Reply:
This is where it gets confusing :)
`qsort` returns a pointer to the type you have passed. In our case, we have passed an array of pointers.
It is legal to have null pointers, but in the compare function, we want to either push the null element to the end, and thus we return `1` or `-1` depends on the null value or return `0` if both are null.